cpp/src/arrow/flight/CMakeLists.txt

```diff
@@ -24,6 +24,10 @@ set(ARROW_FLIGHT_STATIC_LINK_LIBS ${ARROW_PROTOBUF_LIBPROTOBUF} gRPC::grpc++ c-a
 
 if(WIN32)
   list(APPEND ARROW_FLIGHT_STATIC_LINK_LIBS ws2_32.lib)
+elseif(APPLE)
+  if(LIBRESOLV_LIBRARY)
+    list(APPEND ARROW_FLIGHT_STATIC_LINK_LIBS ${LIBRESOLV_LIBRARY})
+  endif()
```

Comment:
Can we use `set_target_properties(c-ares::cares PROPERTIES INTERFACE_LINK_LIBRARIES "${LIBRESOLV_LIBRARY}")` instead of this?

Reply:
Definitely :)

Reply:
Updated and set libresolv as required.